app/services/shepherd.js

```diff
@@ -0,0 +1,145 @@
+import { schedule, later } from "@ember/runloop";
+import Service, { inject as service } from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+import Tours from "timed/tours";
+
+export default class Shepherd extends Service {
+  @service tour;
+  @service notify;
+  @service media;
+  @service router;
+  @service autostartTour;
+  @tracked model;
+
+  get routeName() {
+    return this.router.get("currentRoute")?.name;
+  }
+  async prepare() {
+    this.addDefaultStepsOptions();
+    await this.startFromBeginning();
+    await this.showToursOfCurrentRoute();
+    this.watchRouteChanges();
+  }
+
+  async showToursOfCurrentRoute() {
+    if (this.isThereStepForThisRoute()) {
+      const steps = this.getRouteTours();
+      await this.addSteps(steps.map(this.toStep));
+    }
```

Comment:
We could title those variables and methods a little more precise.
```suggestion
    if (this.isThereStepForThisRoute()) {
      const tours = this.getRouteTours();
       await this.addSteps(tours.map(this.tourToSteps));
    }
```